opentracing-jms-2/src/main/java/io/opentracing/contrib/jms2/TracingJMSProducer.java

```diff
@@ -0,0 +1,433 @@
+        return jmsProducer.getFloatProperty(arg0);
+    }
+
+    @Override
+    public int getIntProperty(String arg0) {
+        return jmsProducer.getIntProperty(arg0);
+    }
+
+    @Override
+    public String getJMSCorrelationID() {
+        return jmsProducer.getJMSCorrelationID();
+    }
+
+    @Override
+    public byte[] getJMSCorrelationIDAsBytes() {
+        return jmsProducer.getJMSCorrelationIDAsBytes();
+    }
+
+    @Override
+    public Destination getJMSReplyTo() {
+        return jmsProducer.getJMSReplyTo();
+    }
+
+    @Override
+    public String getJMSType() {
+        return jmsProducer.getJMSType();
+    }
+
+    @Override
+    public long getLongProperty(String arg0) {
+        return jmsProducer.getLongProperty(arg0);
+    }
+
+    @Override
+    public Object getObjectProperty(String arg0) {
+        return jmsProducer.getObjectProperty(arg0);
+    }
+
+    @Override
+    public int getPriority() {
+        return jmsProducer.getPriority();
+    }
+
+    @Override
+    public Set<String> getPropertyNames() {
+        return jmsProducer.getPropertyNames();
+    }
+
+    @Override
+    public short getShortProperty(String arg0) {
+        return jmsProducer.getShortProperty(arg0);
+    }
+
+    @Override
+    public String getStringProperty(String arg0) {
+        return jmsProducer.getStringProperty(arg0);
+    }
+
+    @Override
+    public long getTimeToLive() {
+        return jmsProducer.getTimeToLive();
+    }
+
+    @Override
+    public boolean propertyExists(String arg0) {
+        return jmsProducer.propertyExists(arg0);
+    }
+
+    @Override
+    public JMSProducer send(Destination destination, Message message) {
+        Span span = TracingMessageUtils.buildAndInjectSpan(destination, message, tracer);
+        try {
+            jmsProducer.send(destination, message);
+        } catch (Throwable e) {
+            SpanJmsDecorator.onError(e, span);
+            throw e;
+        } finally {
+            span.finish();
+        }
+        return this;
+    }
+
+    @Override
+    public JMSProducer send(Destination destination, String message) {
+        TextMessage textMsg = null;
+        try {
+            textMsg = getTextMessage();
+            textMsg.setText(message);
+        } catch (JMSException e) {
+            e.printStackTrace();
+        }
+        return send(destination, textMsg);
+    }
+
+    private TextMessage getTextMessage() throws JMSException{
+        TextMessage textMsg = null;
+        if(jmsContext != null) {
+            textMsg = jmsContext.createTextMessage();
+        }
+        else if (jmsSession != null){
+            textMsg = jmsSession.createTextMessage();
+        }
+        return textMsg;
+    }
+
+    @Override
+    public JMSProducer send(Destination destination, Map<String, Object> arg1) {
+        MapMessage mapMsg = null;
+        try {
+            mapMsg = getMapMessage();
+        } catch (JMSException e) {
+            e.printStackTrace();
```

Comment:
In case of catched exception `mapMsg` will be null.
And we will get NPE later. 
We need to call original `jmsProducer.send(destination, arg1)` and return from method.